### PR TITLE
Run rustfmt on librustc_typeck/check/ folder

### PR DESCRIPTION
**Pull request by srinivasreddy**

### CODE REVIEW

File: src/librustc_typeck/check/writeback.rs [hunk 1]
```diff
@@ -643,10 +650,10 @@ impl<'cx, 'gcx, 'tcx> TypeFolder<'gcx, 'tcx> for Resolver<'cx, 'gcx, 'tcx> {
...
-// promises. It is expected that we will have already reported any
-// errors that may be encountered, so if the promises store an error,
-// a dummy result is returned.
+/// ////////////////////////////////////////////////////////////////////////
+/// During type check, we store promises with the result of trait
+/// lookup rather than the actual results (because the results are not
+/// necessarily available immediately). These routines unwind the
+/// promises. It is expected that we will have already reported any
+/// errors that may be encountered, so if the promises store an error,
+/// a dummy result is returned.
```
Comment (Aatch) [inline src/librustc_typeck/check/writeback.rs]: This isn't correct.

File: src/librustc_typeck/check/cast.rs [hunk 2]
```diff
@@ -145,18 +143,21 @@ impl<'a, 'gcx, 'tcx> CastCheck<'tcx> {
...
-                }, self.expr_ty)
-                    .help(&format!("cast through {} first", match e {
-                            CastError::NeedViaPtr => "a raw pointer",
-                            CastError::NeedViaThinPtr => "a thin pointer",
-                            CastError::NeedViaInt => "an integer",
-                            CastError::NeedViaUsize => "a usize",
-                            _ => bug!()
-                        }))
+                fcx.type_error_struct(self.span,
+                                       |actual| {
```
Comment (jseyfried) [inline src/librustc_typeck/check/cast.rs]: This is misaligned -- rustfmt but?

File: src/librustc_typeck/check/coercion.rs [hunk 3]
```diff
@@ -144,21 +144,18 @@ impl<'f, 'gcx, 'tcx> Coerce<'f, 'gcx, 'tcx> {
...
+                unsize: None,
+            })))
     }
 
-    fn coerce<'a, E, I>(&self,
-                        exprs: &E,
-                        a: Ty<'tcx>,
-                        b: Ty<'tcx>)
-                        -> CoerceResult<'tcx>
-        // FIXME(eddyb) use copyable iterators when that becomes ergonomic.
```
Comment (jseyfried) [inline src/librustc_typeck/check/coercion.rs]: Was this `FIXME` removed on purpose?

**Comment by srinivasreddy** [comment]
@alexcrichton  Can you please stop this merge?

File: src/librustc_typeck/check/coercion.rs [hunk 4]
```diff
@@ -223,9 +220,8 @@ impl<'f, 'gcx, 'tcx> Coerce<'f, 'gcx, 'tcx> {
                                          r_b: &'tcx ty::Region,
                                          mt_b: TypeAndMut<'tcx>)
                                          -> CoerceResult<'tcx>
-        // FIXME(eddyb) use copyable iterators when that becomes ergonomic.
```
Comment (tamird) [inline src/librustc_typeck/check/coercion.rs]: another one

File: src/librustc_typeck/check/coercion.rs [hunk 5]
```diff
@@ -662,9 +654,9 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                            new: &'b hir::Expr,
                                            new_ty: Ty<'tcx>)
                                            -> RelateResult<'tcx, Ty<'tcx>>
-        // FIXME(eddyb) use copyable iterators when that becomes ergonomic.
```
Comment (tamird) [inline src/librustc_typeck/check/coercion.rs]: another

**Pull request merged**